Shared/Fun_GenApp.swift

```diff
@@ -6,12 +6,28 @@
 //
 
 import SwiftUI
+import UIKit
+import Firebase
 
 @main
 struct Fun_GenApp: App {
+    // firebase init
+    @UIApplicationDelegateAdaptor(AppDelegate.self) var appDelegate
+    
     var body: some Scene {
         WindowGroup {
+            let user = UserViewModel()
             ContentView()
+                .environmentObject(user)
```

Comment:
It's generally a bad idea to initialize variables inside `body`. Use it directly instead:

```suggestion
            ContentView()
                .environmentObject(UserViewModel())
```

Reply:
Fixed